src/main/java/com/azure/cosmos/examples/modeling/async/ModelingDemos.java

```diff
@@ -0,0 +1,628 @@
+
+        CosmosPagedFlux<JsonNode> productByCategoryPagedFlux = container.queryItems(
+                "SELECT * FROM c WHERE c.categoryId = '" + categoryId + "'", queryOptions, JsonNode.class);
+
+        productByCategoryPagedFlux.byPage(preferredPageSize).flatMap(cosmosItemPropertiesFeedResponse -> {
+            logger.info("Got a page of query result with " +
+                    cosmosItemPropertiesFeedResponse.getResults().size() + " items(s)"
+                    + " and request charge of " + cosmosItemPropertiesFeedResponse.getRequestCharge());
+
+            for (JsonNode product : cosmosItemPropertiesFeedResponse.getResults()) {
+                ObjectMapper doc = new ObjectMapper();
+                try {
+                    System.out.println(doc.writerWithDefaultPrettyPrinter().writeValueAsString(product));
+                } catch (JsonProcessingException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+            return Flux.empty();
+        }).blockLast();
+
+    }
+
+    public void queryProductsForCategory() {
+        database = client.getDatabase("database-v3");
+        container = database.getContainer("product");
+        int preferredPageSize = 100;
+        CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
+        queryOptions.setQueryMetricsEnabled(true);
+
+        String sql = "SELECT COUNT(1) AS ProductCount, c.categoryName " +
+                "FROM c WHERE c.categoryId = '86F3CBAB-97A7-4D01-BABB-ADEFFFAED6B4' " +
+                "GROUP BY c.categoryName";
+        CosmosPagedFlux<JsonNode> productByCategoryPagedFlux = container.queryItems(
+                sql, queryOptions, JsonNode.class);
+
+        productByCategoryPagedFlux.byPage(preferredPageSize).flatMap(cosmosItemPropertiesFeedResponse -> {
+            logger.info("Got a page of query result with " +
+                    cosmosItemPropertiesFeedResponse.getResults().size() + " items(s)"
+                    + " and request charge of " + cosmosItemPropertiesFeedResponse.getRequestCharge());
+
+            for (JsonNode product : cosmosItemPropertiesFeedResponse.getResults()) {
+                try {
+                    System.out.println("Product count: " + product.get("ProductCount").asText() + "\nCategory name: "
+                            + product.get("categoryName").asText());
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+            return Flux.empty();
+        }).blockLast();
+
+    }
+
+    private void updateProductCategory() {
+        database = client.getDatabase("database-v3");
+        container = database.getContainer("productCategory");
+
+        String categoryId = "86F3CBAB-97A7-4D01-BABB-ADEFFFAED6B4";
+
+        System.out.println("Update the name and replace 'and' with '&'");
```

Comment:
using logger instead?

Reply:
I think the intention of System.out here is more fit for purpose because we really want to print out a message as part of the "demo". In the Deployment.java I agree it was not right.

Reply:
make sense here